src/BackpackServiceProvider.php

```diff
@@ -101,6 +103,35 @@ public function registerMiddlewareGroup(Router $router)
         }
     }
 
+    public function registerMacros()
+    {
+        /**
+         * This macro adds the ability to convert a dot.notation into a [braket][notation] with some special
+         * options that helps us in our usecases.
+         *
+         * - $ignore, usefull when you want to convert a laravel validator rule for nested items and you
+         *   would like to ignore the `*` element from the string.
+         *
+         * - $keyFirst, when true will use the first part of the string as key and only bracket the remaining elements.
+         *   eg: `address.street`
+         *      - when true: `address[street]`
+         *      - when false: `[address][street]`
+         */
+        Str::macro('dotsToSquareBrackets', function ($string, $ignore = [], $keyFirst = false) {
```

Comment:
Bonus: then we wouldn't need to use the third parameter every time we _actually_ call this function 😀

Reply:
What do you think about ignoring `['*']` by default, too? 👀 It would make sense for us, definitely. But... would it make sense for anybody else using this function, for other purposes? What other purpose other than turning `product.*.address.street` to `product[address][street]` is there?

Reply:
Sorry, I can't agree with you on the fact that `key[pair]` is the expected output from a `dotsToSquareBrackets` function.

Why would you expect to not have brackets in one of the keys and have in the others? Of course if you **want to build names for html inputs** you sure want the first element to be the key, but the function is `dotsToSquareBrackets` and not `someDotsToSquareBrackets/createHtmlNameFromDots`, so `something.index === [something][index]`. For our use case, that is: `key => [values]` we want it to be `something[index]` as we will have other `something[create], something[update]`, but that is not what I would expect from a function with that name.

I don't think we should ignore something by default, what I could agree with you is making `keyFirst` true by default since it's OUR function to serve OUR purpose and we can just say, f* it, if somebody uses it and don't want that key input they will find out that there is a parameter to get rid of the behaviour.

Reply:
Ok, fair enough. Let's default to `true`, we do not ignore anything by default.